GoMeet/src/main/webapp/js/manual-time-proposal.js

```diff
@@ -0,0 +1,161 @@
+const MAX_TIME_INPUTS = 5; // Allow up to 5 proposed times
+var timeInputs = 1; // Current number of time inputs on page
+const MEETING_TIME_NAME_PREFIX = 'meeting-time-'; // The prefix of the sessionStorage key name for meeting times
+var timeNameSuffix = 1; // The suffix appended to MEETING_TIME_NAME_PREFIX to make the name unique
+const MAX_TIME_INPUT_REACHED_ALERT = 'Maximum number of proposed times: ' + MAX_TIME_INPUTS.toString();
+const TIME_INPUT_FIELD_INDEX = 1; // The index of the <input> field with its parent div
+var enteredTimes = new Set();
+
+/**
+ * Called onload to set the minimum time users can input into the <input> with
+ * name 'meeting-time-1' (default input visible on page load) to the current 
+ * date and time.
+ * @param {Document} document the document that this function operates on.
+ */
+function setMinDatetime(document) {
+  document.getElementsByName('meeting-time-1').item(0).min = getDatetimeNow();
+}
+
+/**
+ * Adds a time input field and corresponding delete button to the meeting-time-inputs div.
+ * @param {Document} document the document that this function operates on.
+ */
+function addTimeInput(document) {
+  if (timeInputs >= MAX_TIME_INPUTS) {
+    alert(MAX_TIME_INPUT_REACHED_ALERT);
+    return;
+  }
+
+  timeInputs++
+  // Show delete buttons when it has reached 2 time inputs on the page
+  if (timeInputs === 2) {
+    toggleDeleteButtons(document);
+  }
+  timeNameSuffix++
+  
+  let inputDiv = document.getElementById('meeting-time-inputs');
+  let inputField = document.createElement('input');
+  inputField.type = 'datetime-local';
+  inputField.name = MEETING_TIME_NAME_PREFIX + timeNameSuffix.toString();
+  // restrict potential meeting times to future dates only
+  let minDate = getDatetimeNow();
+  inputField.min = minDate;
+  // whenever user changes the input value, check what the time they inputted to be valid
+  inputField.onchange = function() {rectifyInputtedTime(this);};
+
+  let label = document.createElement('label');
+  label.htmlFor = inputField.name;
+  label.innerText = 'New Time';
+
+  let deleteButton = document.createElement('button');
+  deleteButton.innerText = '-';
+  deleteButton.className = 'delete-time-button';
+  deleteButton.onclick = function() {deleteTimeInput(document, inputField.name);}; 
+  
+  let enclosingDiv = document.createElement('div');
+  enclosingDiv.id = inputField.name; // ID of div = name of input field, to faciliate deleting
+
+  enclosingDiv.appendChild(label);
+  enclosingDiv.appendChild(inputField);
+  enclosingDiv.appendChild(deleteButton);
+  inputDiv.appendChild(enclosingDiv);
```

Comment:
Optional suggestion: Maybe you can use a function to create enclosingDiv.

Reply:
I will add that to my refactoring list 👍

Reply:
Again, this was really quick, and does remove repeated code, so I've refactored it 😊